crates/ruff_benchmark/benches/red_knot.rs

```diff
@@ -23,14 +25,52 @@ struct Case {
 
 const TOMLLIB_312_URL: &str = "https://raw.githubusercontent.com/python/cpython/8e8a4baf652f6e1cee7acde9d78c4b6154539748/Lib/tomllib";
 
-static EXPECTED_DIAGNOSTICS: &[&str] = &[
+/// A structured set of fields we use to do diagnostic comparisons.
+///
+/// This helps assert benchmark results. Previously, we would compare
+/// the actual diagnostic output, but using `insta` inside benchmarks is
+/// problematic, and updating the strings otherwise when diagnostic rendering
+/// changes is a PITA.
+type KeyDiagnosticFields = (
```

Comment:
I think we may want `.message()` string in here too, but I'm not sure how much difference it will make going forward, and we can always add it back in as a separate change.

Reply:
Done!

Can you say more about why `.message()` is desirable here? Do you mean it's desirable to test and validate, or do you mean it's desirable from a measurement perspective?

Reply:
It's desirable from a usability perspective of validating in PR review that the diagnostics removed from (or added to) this assertion list are sensibly related to the red-knot changes.

Reply:
Ah I think I understand now. Thank you for explaining!